lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -703,8 +704,9 @@ protected function get_allowed_query_vars() {
 		 * Allows adjusting of the default query vars that are made public.
 		 *
 		 * @param array  Array of allowed WP_Query query vars.
+		 * @param WP_REST_Request $request Request object.
 		 */
-		$valid_vars = apply_filters( 'query_vars', $wp->public_query_vars );
+		$valid_vars = apply_filters( 'query_vars', $wp->public_query_vars, $request );
```

Comment:
This is a [hook in core](https://developer.wordpress.org/reference/hooks/query_vars/) which we're duplicating here for technical reasons, so I'm not sure we _can_ pass the request here. Is `rest_query_vars` not powerful enough for this?